### PR TITLE
gh-101640: Make argparse working in `pythonw` by dropping stderr writes there

### PR DESCRIPTION
**Pull request by arhadthedev**
A note on freshly added tests:

- `StdStreamTest.test_skip_invalid_stderr()` is confirmed to be failing before the patch
- `StdStreamTest.test_skip_invalid_stdout()` is added for symmetry; its tested behavior already worked correctly

* Issue: gh-101640

### CODE REVIEW

**Comment by arhadthedev** [comment]
Ping?

**Comment by arhadthedev** [comment]
ping

File: Lib/argparse.py [hunk 1]
```diff
@@ -2605,9 +2605,9 @@ def print_help(self, file=None):
 
     def _print_message(self, message, file=None):
         if message:
-            if file is None:
-                file = _sys.stderr
-            file.write(message)
+            file = file or _sys.stderr
+            if file is not None:
+                file.write(message)
```
Comment (terryjreedy) [inline Lib/argparse.py]: Catch issues with unwritable non-None files with
```suggestion
            try:
                file.write(message)
            except (AttributeError, OSError):
                pass
```

File: Misc/NEWS.d/next/Library/2023-02-09-22-24-34.gh-issue-101640.oFuEpB.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+:class:`argparse.ArgumentParser` no longer tries to use :data:`sys.stderr`
+when it is set to ``None``. Patch by Oleg Iarygin.
```
Comment (terryjreedy) [inline Misc/NEWS.d/next/Library/2023-02-09-22-24-34.gh-issue-101640.oFuEpB.rst]: If you accept the suggestion above:
```suggestion
:class:`argparse.ArgumentParser` now catches errors when writing messages, such as when :data:`sys.stderr` is ``None``. Patch by Oleg Iarygin.
```

**Review by terryjreedy — CHANGES_REQUESTED** [review]
I am not sure who you are pinging, but... tests look OK to me.  I just suggest catching any write error.

**Comment by arhadthedev** [comment]
Retriggering the build to restart the failed `docs/readthedocs.org:cpython-previews`:

```plain
Running Sphinx v4.5.0
loading translations [en]... done

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/cpython-previews/envs/101802/lib/python3.7/site-packages/sphinx/registry.py", line 425, in load_extension
    mod = import_module(extname)
  File "/home/docs/checkouts/readthedocs.org/user_builds/cpython-previews/envs/101802/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/docs/checkouts/readthedocs.org/user_builds/cpython-previews/envs/101802/lib/python3.7/site-packages/sphinx/builders/linkcheck.py", line 20, in <module>
    from requests import Response
  File "/home/docs/checkouts/readthedocs.org/user_builds/cpython-previews/envs/101802/lib/python3.7/site-packages/requests/__init__.py", line 43, in <module>
    import urllib3
  File "/home/docs/checkouts/readthedocs.org/user_builds/cpython-previews/envs/101802/lib/python3.7/site-packages/urllib3/__init__.py", line 39, in <module>
    "urllib3 v2.0 only supports OpenSSL 1.1.1+, currently "
ImportError: urllib3 v2.0 only supports OpenSSL 1.1.1+, currently the 'ssl' module is compiled with OpenSSL 1.0.2n  7 Dec 2017. See: https://github.com/urllib3/urllib3/issues/2168
```

**Comment by arhadthedev** [comment]
The docs deploy continues to fail. However, this PR doesn't change documentation so I believe it can be ignored for now.

For Bedevere: I have made the requested changes; please review again.

**Review by terryjreedy — APPROVED** [review]

**Pull request merged**